src/Analyser/TypeSpecifier.php

```diff
@@ -466,6 +426,26 @@ public function specifyTypesInCondition(
 				}
 			}
 
+			if (
+				!$context->null()
+				&& $expr->right instanceof Expr\FuncCall
+				&& $expr->right->name instanceof Name
+				&& in_array(strtolower((string) $expr->right->name), ['preg_match', 'strlen', 'mb_strlen'], true)
+			) {
```

Comment:
atm I whitelisted the cases which I dropped above to keep my sanity.
I feel we can drop it after we removed the hard-coded cases and moved them into extensions